src/elements/components/download-button.tsx

```diff
@@ -0,0 +1,70 @@
+import { FiExternalLink } from 'react-icons/fi';
+import { Resource } from '../../lib/schema';
+
+type DownloadButtonProps = {
+    url: string;
+    resource: Resource;
+};
+
+const getHostFromUrl = (url: string) => {
+    if (url.includes('github')) {
+        return 'GitHub';
+    }
+    if (url.includes('drive.google')) {
+        return 'Google Drive';
+    }
```

Comment:
I can only see this go wrong long term. Could you parse the URL and look at the domain instead?

Reply:
that's a good point lol..... `www.drive.google.suslink.no/dont-click-this.exe`